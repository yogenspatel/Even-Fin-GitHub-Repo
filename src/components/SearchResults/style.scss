.search-container {
    margin: 50px auto;
    overflow-y: scroll;
}

.search-container .text-center {
    margin-top: 50px;
}

.search-results .row {
    border: 1px solid #ccc;
    margin-bottom: 10px;
}

.search-results .row .col-6 .alert-light {
    padding-top: 15px;
    margin-bottom: 10px;
}

.search-results .row .col:first-child {
    border-left: 1px solid #ccc;
    border-right: 1px solid #ccc;
}

.search-results .row .col-6 a, .search-results .row .col span {
    margin-top: 10px;
    display: block;
}
