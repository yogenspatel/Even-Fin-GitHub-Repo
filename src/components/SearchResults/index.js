import React from "react";
import "./style.scss";

const SearchResults = ({ results } = null) => {
  if (results && results.length) {
    const resultsItems = results.map(item => (
      <div className="row" key={item.id}>
        <div className="col-6">
          <a href={item.url} target="_blank" rel="noopener noreferrer">
            {item.full_name}
          </a>
          {item.fork && <div className="fork">Forked</div>}
          <p className="alert-light">{item.description}</p>
        </div>
        <div className="col">
          <span className="alert-light">Stars:</span>
          <p className="h5 stars">{item.stargazers_count}</p>
        </div>
        <div className="col">
          <span className="alert-light">License</span>
          {item.license && item.license.name && <p className="h5 license">{item.license.name}</p>}
        </div>
      </div>
    ));
    return (
      <div className="container search-container">
        <hr />
        <p className="col text-center alert-light">SEARCH results</p>
        <div className="search-results">{resultsItems}</div>
      </div>
    );
  }

  return <div>No Search Results found</div>;
};

export default SearchResults;
