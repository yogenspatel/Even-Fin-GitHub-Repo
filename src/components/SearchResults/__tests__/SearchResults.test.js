/* eslint-disable no-undef */

import React from "react";
import ReactDOM from "react-dom";
import SearchResults from "../../SearchResults";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

it("SearchResults renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<SearchResults />, div);
});

it("SearchResults Match Snapshot", () => {
  const wrapper = shallow(<SearchResults />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
