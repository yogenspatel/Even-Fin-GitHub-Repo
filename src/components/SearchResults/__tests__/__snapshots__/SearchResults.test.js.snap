// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults Match Snapshot 1`] = `
<div>
  No Search Results found
</div>
`;
